src/guidellm/scheduler/constraints/__init__.py

```diff
@@ -0,0 +1,51 @@
+"""
+Constraint system for scheduler behavior control and request processing limits.
+
+Provides flexible constraints for managing scheduler behavior with configurable
+thresholds based on time, error rates, and request counts. Constraints evaluate
+scheduler state and individual requests to determine whether processing should
+continue or stop based on predefined limits. The constraint system enables
+sophisticated benchmark stopping criteria through composable constraint types.
+"""
+
+from .base import (
```

Comment:
can we merge this and protocols into a single file called `constraint.py` to better merge some other new patterns across the repo?